ROADMAP.md

```diff
@@ -0,0 +1,155 @@
+# js-libp2p roadmap Q4’22/Q1’23  <!-- omit in toc -->
+
+```
+Date: 2022-10-07
+Status: In Progress
+Notes: This document is still in review will may be heavily modified based on stakeholder feedback. Please add any feedback or questions in:
+https://github.com/libp2p/js-libp2p/issues/1438
+```
+
+## Table of Contents <!-- omit in toc -->
+
+- [About the Roadmap](#about-the-roadmap)
+	- [Vision](#vision)
+	- [Sections](#sections)
+	- [Done criteria](#done-criteria)
+	- [Benchmarking and Testing](#benchmarking-and-testing)
+- [🛣️ Milestones](#️-milestones)
+	- [2022](#2022)
+		- [Early Q4 (October)](#early-q4-october)
+		- [Mid Q4 (November)](#mid-q4-november)
+		- [End of Q4 (December)](#end-of-q4-december)
+	- [2023](#2023)
+		- [Mid Q1 (February)](#mid-q1-february)
+		- [End of Q1 (March)](#end-of-q1-march)
+		- [Early Q2 (April)](#early-q2-april)
+	- [Up Next](#up-next)
+- [📖 Appendix](#-appendix)
+	- [A. 📺 Universal Browser Connectivity](#a--universal-browser-connectivity)
+		- [1. WebTransport](#1-webtransport)
+		- [2. WebRTC: Browser to Server](#2-webrtc-browser-to-server)
+		- [3. WebRTC: Browser to Browser](#3-webrtc-browser-to-browser)
+	- [B. 🥊 Decentralized Hole Punching](#b--decentralized-hole-punching)
+		- [1. AutoNat](#1-autonat)
+		- [2. Circuit Relay v2](#2-circuit-relay-v2)
+		- [3. Add QUIC Transport](#3-add-quic-transport)
+		- [4. Hole Punching](#4-hole-punching)
+	- [C. 🔮 Future-proof Observability](#c--future-proof-observability)
+		- [1. Benchmark transfer performance in CI](#1-benchmark-transfer-performance-in-ci)
+		- [2. Add introspection to js-libp2p](#2-add-introspection-to-js-libp2p)
+- [libp2p Project Roadmap](#libp2p-project-roadmap)
+
+## About the Roadmap
+
+### Vision
+The tenets of the libp2p team are:
+- Security
+  - We strive to ensure that libp2p is reliable & secure for users & applications. We seek to minimize vulnerabilities and act quickly to address issues.
+- Stability
+  - We do not break deployed features and maintain cross version compatibility across libp2p releases.
+- Performance
+  - We aim to make libp2p as performant as possible and strive to ensure comparable performance to widely used protocols on the Internet. We avoid regressions and performance degradations in new releases.
```

Comment:
Per discussion in go-libp2p, lets reduce the duplication and link to an authoritative discussion in specs ROADMAP.